### DISCUSSION TITLE
Slack and Google Auth Problem

### DISCUSSION THREAD

**Discussion by udoyhasan** [outline/outline#3151] [Self Hosting] [answered]
I'm new to Outline. I've been trying to install our line on my dedicated server. I've tried to install outline from scratch but it was giving me an error while installing. So, I've installed it successfully from the source with all requirements. But after starting I'm having a problem.

The problem is:
I've installed the site on outline.example.com so when I try to login via slack it redirects me to `outline.example.com?notice=authentication-required` when login is successful. The same thing also happens when I use google auth. I've tried to debug and found that the following errors are occurring while I try to log in.

Slack Error:
```bash
Error during authentication {
  error: ClientError [UnauthorizedError]: Decorating class property failed. Please ensure that proposal-class-properties is enabled and runs after the decorators transform.
      at AuthenticationError (/path/to/outline/build/server/errors.js:33:34)
      at accountProvisioner (/path/to/outline/build/server/commands/accountProvisioner.js:42:43)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Strategy._verify (/path/to/outline/build/server/routes/auth/providers/slack.js:59:22) {
    redirectUrl: 'https://outline.example.com',
    id: 'authentication_required'
  },
  extra: undefined
}
```

Google Error:
```bash
Error during authentication {
  error: ClientError [UnauthorizedError]: Decorating class property failed. Please ensure that proposal-class-properties is enabled and runs after the decorators transform.
      at AuthenticationError (/path/to/outline/build/server/errors.js:33:34)
      at accountProvisioner (/path/to/outline/build/server/commands/accountProvisioner.js:42:43)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async Strategy._verify (/path/to/outline/build/server/routes/auth/providers/google.js:65:22) {
    redirectUrl: 'https://outline.example.com',
    id: 'authentication_required'
  },
  extra: undefined
}
```

I hope if anybody can help me :)

**Comment by tommoor** [comment]
This is a babel error from code transpilation. We'd need to know more details – which steps did you go through to compile the code and what are the versions of all dependencies.

I _highly_ recommend you use the official docker image instead.

**Comment by udoyhasan** [comment]
@tommoor thank you very much for your reply. I've followed this tutorial https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/from-source-BlBxrNzMIP

And my setup is:
NodeJS: 16.x LTS
Yarn: Latest
Postgresql: Latest
Redis: Latest


About the docker, the official docs about docker give me an error so I tried to install it from the source.

**Comment by udoyhasan** [comment, marked as answer]
I've solved the problem by updating to v0.61.1
